nin/dasBoot/NodeManager.js

```diff
@@ -0,0 +1,132 @@
+class NodeManager {
+
+  constructor() {
+    this.nodes = {};
+    this.graphChangeListeners = [];
+  }
+
+  onGraphChange(fn) {
+    this.graphChangeListeners.push(fn);
+  }
+
+  fireGraphChange() {
+    console.log('graph change!');
+    for(var i = 0; i < this.graphChangeListeners.length; i++) {
+      this.graphChangeListeners[i]();
+    }
+  }
+
+  createNode(nodeInfo) {
+    nodeInfo.options = nodeInfo.options || {};
+    var node;
+    if(nodeInfo.type.slice(0, 4) == 'NIN.') {
+      node = new NIN[nodeInfo.type.slice(4)](nodeInfo.id, nodeInfo.options);
+    } else {
```

Comment:
The check isn't for whether or not they inherit from NIN.Node. All nodes inherit from NIN.Node. The check is for whether or not the class is defined in the NIN namespace. An example of a node class that is defined outside of the NIN namespace is a node type that is defined in a demo project, e.g. MyAwesomeRotatingCubeSceneNode.